zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -289,12 +291,34 @@ impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
                     }
                 }
 
-                add_account_internal::<P, WalletMigrationError>(
-                    &self.params,
-                    transaction,
-                    "accounts_new",
-                    account,
-                    &ufvk,
+                let ufvk_str: String = ufvk.encode(&self.params);
+                let address_str: String = ufvk.default_address().0.encode(&self.params);
+
+                // This migration, and the wallet behaviour before it, stored the default
+                // transparent address in the `accounts` table. This does not necessarily
+                // match the transparent receiver in the default Unified Address. Starting
+                // from `AddressesTableMigration` below, we no longer store transparent
+                // addresses directly, but instead extract them from the Unified Address
+                // (or from the UFVK if the UA was derived without a transparent receiver,
+                // which is not the case for UAs generated by this crate).
```

Comment:
```suggestion
                // addresses directly, but instead extract them from the UFVK.
```
As far as I can see the code does always extract them from the UFVK, regardless of whether they exist in the UA.